crates/fluvio-dataplane-protocol/src/batch.rs

```diff
@@ -360,7 +360,7 @@ pub struct BatchHeader {
     pub partition_leader_epoch: i32,
     pub magic: i8,
     pub crc: u32,
-    pub attributes: i16,
+    pub attributes: u16,
```

Comment:
Is there an specific reason to make it `u16`. I think that this change is good to have and probably harmless but not sure if this should be in this PR

Reply:
I agree, let's reduce changes

Reply:
Yeah, this change wasn't a must, it just made more sense to deal with unsigned values in combination with shifting and masking operations. The choice of signed int also propagates to conversions between `Compression` and primitive values.

I'll revert it to keep the PR focused.